### DISCUSSION TITLE
C64 IO_Monitor patch for Durexforth

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#388] [Show and tell]
https://github.com/davervw/c64-io_monitor

 It allows $1f00 bytes of logging. This can be increased if anyone is interested.

```forth
hex
9 device
parse-name io?monitor
c000 loadb
drop 8 device

 ea eaea c009 ! c00b c!
\ banknorm

384c c025 ! c0 c027 c!
\ copy basic

a9 c040 !
\ log ptr

e9 c048 !
\ log rem

a0a9 c04d !
\ msb ptr

bf c053 c!
\ v buff

a0e9 c054 !

a9 c0f0 !
a0a9 c0f4 !
\ displaylog

```
![vice-screen-20211224013407](https://user-images.githubusercontent.com/38296423/147345796-352e657e-6695-4e37-abbb-62aeab70edda.jpg)
![vice-screen-20211224114530](https://user-images.githubusercontent.com/38296423/147370598-2819c826-70ff-4cda-8fd8-b09fe3ba1301.jpg)

**Comment by Whammo** [comment]
And a few words to make it nice and useful.

```forth


: init c000 sys ;
: cnt c003 sys ;
: log c006 sys ;
: logend ( msb - )
$c053 c! ;
: logstart ( msb - )
dup dup $c04e c! $c055 c! $c0f5 c!
;

\ here 100/ 1+ logstart
\ 6f logend

: slog s" @0:log,p,w"
8 8 open ioabort clrchn
8 chkout ioabort 0 emit 0 emit
cnt log clrchn 8 close ;

```

**Reply by polluks** [reply]
The term "monitor" is misleading, it's a profiler IMHO.

**Reply by Whammo** [reply]
I'd like to see it do the low level IEC calls, but the Kernal doesn't use the vectors.
